src/include/duckdb/main/client_config.hpp

```diff
@@ -23,6 +23,53 @@ class PreparedStatementData;
 typedef std::function<unique_ptr<PhysicalResultCollector>(ClientContext &context, PreparedStatementData &data)>
     get_result_collector_t;
 
+enum class TreeNodeSettingsType : uint8_t { CPU_TIME, EXTRA_INFO, OPERATOR_CARDINALITY, OPERATOR_TIMING };
+
+const unordered_map<TreeNodeSettingsType, Value> default_metrics = {
+    {TreeNodeSettingsType::CPU_TIME, Value()},
+    {TreeNodeSettingsType::EXTRA_INFO, Value()},
+    {TreeNodeSettingsType::OPERATOR_CARDINALITY, Value()},
+    {TreeNodeSettingsType::OPERATOR_TIMING, Value()},
+};
+
+class TreeNodeSettings {
+private:
+	// map of metrics with their values; only enabled metrics are present in the map
+	unordered_map<TreeNodeSettingsType, Value> metrics = default_metrics;
+
+public:
+	TreeNodeSettings() = default;
+	TreeNodeSettings(TreeNodeSettings &) = default;
+	TreeNodeSettings &operator=(TreeNodeSettings &) = default;
+
+	void set_setting(const TreeNodeSettingsType &setting, const Value &value) {
```

Comment:
Nit: duckdb has CamelCase function names. Also, passing a `uint8_t` enum as a reference copies 8 bytes instead of 1.

Reply:
All enums in duckdb seem to be `uint8_t`.